webpack/scenes/ContentViews/Table/tableDataGenerator.js

```diff
@@ -0,0 +1,132 @@
+    lastPublished || 'Not yet published',
+    { title: __('Details'), props: { isOpen: false, ariaControls: `cv-details-expansion-${id}` } },
+    {
+      title: <IconWithCount Icon={ScreenIcon} count={environments.length} title={`environments-icon-${id}`} />,
+      props: { isOpen: false, ariaControls: `cv-environments-expansion-${id}` },
+    },
+    {
+      title: <IconWithCount Icon={RepositoryIcon} count={repositories.length} title={`repositories-icon-${id}`} />,
+      props: { isOpen: false, ariaControls: `cv-repositories-expansion-${id}` },
+    },
+    {
+      title: <IconWithCount Icon={ContainerNodeIcon} count={versions.length} title={`versions-icon-${id}`} />,
+      props: { isOpen: false, ariaControls: `cv-versions-expansion-${id}` },
+    },
+  ];
+  if (openColumn) row[openColumn].props.isOpen = true;
+
+  return row;
+};
+
+const buildDetailDropdowns = (id, rowIndex, openColumn) => {
+  const cvId = { cvId: id };
+  const expansionProps = { ...cvId, className: 'pf-m-no-padding' };
+  const containerProps = column => ({ ...cvId, isOpen: openColumn === column });
+
+  let detailDropdowns = [
+    {
+      compoundParent: 2,
+      cells: [
+        {
+          title: (
+            <DetailsContainer {...containerProps(3)}>
+              <DetailsExpansion {...expansionProps} />
+            </DetailsContainer>),
+          props: { colSpan: 6 },
+        },
+      ],
+    },
+    {
+      compoundParent: 3,
+      cells: [
+        {
+          title: (
+            <DetailsContainer {...containerProps(4)}>
+              <EnvironmentsExpansion {...expansionProps} />
+            </DetailsContainer>),
+          props: { colSpan: 6 },
+        },
+      ],
+    },
+    {
+      compoundParent: 4,
+      cells: [
+        {
+          title: (
+            <DetailsContainer {...containerProps(5)}>
+              <RepositoriesExpansion {...expansionProps} />
+            </DetailsContainer>),
+          props: { colSpan: 6 },
+        },
+      ],
+    },
+    {
+      compoundParent: 5,
+      cells: [
+        {
+          title: (
+            <DetailsContainer {...containerProps(6)}>
+              <VersionsExpansion {...expansionProps} />
+            </DetailsContainer>),
+          props: { colSpan: 6 },
+        },
+      ],
+    },
+  ];
+
+  // The rows are indexed along with the hidden dropdown rows, so we need to offset the parent row
+  const rowOffset = detailDropdowns.length + 1;
+  detailDropdowns = detailDropdowns.map(detail => ({ ...detail, parent: rowIndex * rowOffset }));
+
+  return detailDropdowns;
+};
+
+const tableDataGenerator = (results, rowMapping) => {
+  const updatedRowMapping = { ...rowMapping };
+  const contentViews = results || [];
+  const columns = buildColumns();
+  const rows = [];
+
+  contentViews.forEach((contentView, rowIndex) => {
+    const { id } = contentView;
+    if (!Object.prototype.hasOwnProperty.call(updatedRowMapping, id)) {
```

Comment:
I'm very confused.. why do you need `call` here?  And why use `Object.prototype` ?

Reply:
eslint yelled at me! https://eslint.org/docs/rules/no-prototype-builtins

Reply:
Wow.  I can see that it's a reasonable rule but it sure makes the code hard to read and understand.  Thanks for explaining :)

Reply:
Actually I would like a comment pointing to that eslint rule.